### PR TITLE
[DROOLS-7042] Configure invoker to output in junit format

### PR DESCRIPTION
**Pull request by gitgabrio**
@radtriste @winklerm 

**JIRA**: 

[link](https://issues.redhat.com/browse/DROOLS-7042)

With this configuration, the invoker plugin write ALSO `TEST-...` files in junit format, inside `invoker-reports`

See https://github.com/jenkinsci/maven-invoker-plugin/blob/master/README.md for jenkins junit configuration

This is the beginning of the generated `TEST-...` file

`<testsuite tests="1" failures="0" name="org.kie.maven.plugin.it.kie-maven-plugin-test-kjar-setup" time="6.103" errors="0" skipped="0">
  <testcase classname="org.kie.maven.plugin.it.kie-maven-plugin-test-kjar-setup" name="kie-maven-plugin-test-kjar-setup" time="6.103">
    <system-out>Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)`

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>
</details>

### CODE REVIEW

File: kie-maven-plugin/pom.xml [hunk 1]
```diff
@@ -17,6 +17,15 @@
   <name>KIE :: Maven Plugin</name>
 
   <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-invoker-plugin</artifactId>
+          <version>3.2.1</version>
```
Comment (radtriste) [inline kie-maven-plugin/pom.xml]: I guess we should manage that version for the whole project in the drools-build-parent ?

File: kie-maven-plugin/pom.xml [hunk 2]
```diff
@@ -25,12 +34,14 @@
...
       </plugin>
-       <plugin>
+      <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-invoker-plugin</artifactId>
         <configuration>
           <skipInstallation>${skipITs}</skipInstallation>
           <skipInvocation>${skipITs}</skipInvocation>
+          <writeJunitReport>false</writeJunitReport>
+          <junitPackageName></junitPackageName>
```
Comment (radtriste) [inline kie-maven-plugin/pom.xml]: could we use a proper packageName, like `org.kie.maven.plugin.it` ?

**Comment by radtriste** [comment]
Please also update https://github.com/kiegroup/drools/blob/main/.ci/jenkins/Jenkinsfile.deploy#L193 and https://github.com/kiegroup/drools/blob/main/.ci/jenkins/Jenkinsfile.native#L55 by adding the `**/target/invoker-reports/**/*.xml`

File: kie-maven-plugin/pom.xml (hunk 2 above)
Comment (gitgabrio) [reply kie-maven-plugin/pom.xml]: @radtriste 

Done :+1:

File: kie-maven-plugin/pom.xml (hunk 1 above)
Comment (gitgabrio) [reply kie-maven-plugin/pom.xml]: @radtriste 
done

**Comment by gitgabrio** [comment]
> Please also update https://github.com/kiegroup/drools/blob/main/.ci/jenkins/Jenkinsfile.deploy#L193 and https://github.com/kiegroup/drools/blob/main/.ci/jenkins/Jenkinsfile.native#L55 by adding the `**/target/invoker-reports/**/*.xml`

@radtriste 
Done

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
https://github.com/kiegroup/kogito-pipelines/pull/522 has been merged. testing the changes to get the invoker reports as junit in Jenkins.

jenkins rerun drools tests

**Comment by radtriste** [comment]
GHA kogito-runtimes should be solved by https://github.com/kiegroup/kogito-runtimes/pull/2299

**Comment by radtriste** [comment]
invoker reports are generated correctly as junit results

**Pull request merged**